src/modules/pm.c

```diff
@@ -142,26 +174,51 @@ static void publish_susp_req(const bool new) {
     M_PUB(suspend_req);
 }
 
-static void on_pm_req(const bool new) {
-    if (new && pm_inh_token == -1) {
-        USERBUS_ARG_REPLY(pm_args, parse_bus_reply, &pm_inh_token,
-                            "org.freedesktop.PowerManagement.Inhibit",
-                            "/org/freedesktop/PowerManagement/Inhibit",
-                            "org.freedesktop.PowerManagement.Inhibit",
-                            "Inhibit");
-        if (call(&pm_args, "ss", "Clight", "PM inhibition.") == 0) {
-            DEBUG("Holding PowerManagement inhibition.\n");
+static bool acquire_lock(void) {
+    static bool r = false;
+    if (pm_inh_token == -1 || ext_inhibit) {
+        SYSBUS_ARG_REPLY(pm_args, parse_bus_reply, &pm_inh_token, "org.freedesktop.login1", "/org/freedesktop/login1",
+                         "org.freedesktop.login1.Manager", "Inhibit");
+
+        int ret = call(&pm_args, "ssss", "idle", "Clight", "Idle inhibitor.", "block");
+
+        if (ret < 0) {
+            DEBUG("Failed to parse D-Bus response for idle inhibit\n");
+        }
+
+        if (pm_inh_token < 0) {
+            DEBUG("Failed to copy lock file\n");
+        } else
```

Comment:
Missing braces

Reply:
I dont think braces are specifically needed here since there is only one line, but I'll add for clarity.